### PR TITLE
azidentity: Add service fabric test program

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding code to test in service fabric for #12692

### CODE REVIEW

**Comment by catalinaperalta** [comment]
Pending azidentity version update.

File: sdk/samples/azidentity/manual-tests/managed-identity/service-fabric/go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module main
+
+go 1.15
+
+require (
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.14.0
```
Comment (jhendrixMSFT) [inline sdk/samples/azidentity/manual-tests/managed-identity/service-fabric/go.mod:6]: This isn't the latest version, will that have any impact?

**Review by jhendrixMSFT — APPROVED** [review]
One comment regarding old `azcore`

File: sdk/samples/azidentity/manual-tests/managed-identity/service-fabric/go.mod (hunk 1 above)
Comment (catalinaperalta) [reply sdk/samples/azidentity/manual-tests/managed-identity/service-fabric/go.mod]: Done, updated to latest.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**